### ISSUE TITLE
Tests 3.9 to 3.14 are failing to find certificate files

### ISSUE THREAD

**Issue by kinghuang** [docker/docker-bench-security#232]
**Description**

Tests 3.9 to 3.14 are failing to find the certificate files they are checking for in my environment. They are reporting no certificate/key found, when the files actually exist.

**Steps to reproduce the issue:**

Given the following `daemon.json` configuratation and files:

```
{
	"disable-legacy-registry": true,
	"experimental": true,
	"storage-driver": "devicemapper",
	"storage-opts" : [
		"dm.thinpooldev=/dev/mapper/docker-thinpool",
		"dm.use_deferred_removal=true"
	],
	"hosts": [
		"unix:///var/run/docker.sock",
		"tcp://0.0.0.0:2376"
	],
	"metrics-addr": "0.0.0.0:9323",
	"tlsverify": true,
	"tlscacert": "/etc/docker/certs.d/ca.pem",
	"tlscert": "/etc/docker/certs.d/server-cert.pem",
	"tlskey": "/etc/docker/certs.d/server-key.pem"
}
```

```
-bash-4.2$ ls -al /etc/docker/certs.d
total 36
drwxr-xr-x. 2 root root 4096 Jan 10 15:21 .
drwx------. 3 root root   53 Feb 20 12:17 ..
-r--------. 1 root root 3326 Jan 10 15:21 ca-key.pem
-r--r--r--. 1 root root 2240 Jan 10 15:21 ca.pem
-rw-r--r--. 1 root root   17 Jan 10 15:21 ca.srl
-r--r--r--. 1 root root 1931 Jan 10 15:21 cert.pem
-rw-r--r--. 1 root root   30 Jan 10 15:21 extfile.cnf
-r--------. 1 root root 3247 Jan 10 15:21 key.pem
-r--r--r--. 1 root root 2086 Jan 10 15:21 server-cert.pem
-r--------. 1 root root 3243 Jan 10 15:21 server-key.pem
```

1. Run `docker-bench-security.sh`.

**Describe the results you received:**

Tests 3.9 to 3.14 all failed to find the configured certificate and key files.

```
[INFO] 3.9  - Verify that TLS CA certificate file ownership is set to root:root
[INFO]      * No TLS CA certificate found
[INFO] 3.10 - Verify that TLS CA certificate file permissions are set to 444 or more restrictive
[INFO]      * No TLS CA certificate found
[INFO] 3.11 - Verify that Docker server certificate file ownership is set to root:root
[INFO]      * No TLS Server certificate found
[INFO] 3.12 - Verify that Docker server certificate file permissions are set to 444 or more restrictive
[INFO]      * No TLS Server certificate found
[INFO] 3.13 - Verify that Docker server key file ownership is set to root:root
[INFO]      * No TLS Key found
[INFO] 3.14 - Verify that Docker server key file permissions are set to 400 or more restrictive
[INFO]      * No TLS Key found
```

**Describe the results you expected:**

The files defined in `tlscacert`, `tlscert`, and `tlskey` in `daemon.json` should have been found and their ownership and permissions verified.

**Additional information you deem important (e.g. issue happens only occasionally):**

This was tested with the `v1.3.1` release.

**Comment by konstruktoid** [comment]
Hi @kinghuang, thanks for reporting this. I will look into it and update as soon as possible.

**Comment by konstruktoid** [comment]
Please test https://github.com/docker/docker-bench-security/pull/234

**Comment by kinghuang** [comment]
@konstruktoid On #234, it's detecting the certificate files and their ownership correctly (3.9, 3.11, 3.13). But, it's failing on the permission checks (3.10, 3.12, 3.14).

```
[PASS] 3.9  - Verify that TLS CA certificate file ownership is set to root:root
[WARN] 3.10 - Verify that TLS CA certificate file permissions are set to 444 or more restrictive
[WARN]      * Wrong permissions for /etc/docker/certs.d/ca.pem
[PASS] 3.11 - Verify that Docker server certificate file ownership is set to root:root
[WARN] 3.12 - Verify that Docker server certificate file permissions are set to 444 or more restrictive
[WARN]      * Wrong permissions for /etc/docker/certs.d/server-cert.pem
[PASS] 3.13 - Verify that Docker server key file ownership is set to root:root
[WARN] 3.14 - Verify that Docker server key file permissions are set to 400 or more restrictive
[WARN]      * Wrong permissions for /etc/docker/certs.d/server-key.pem
```

```
[root@itpocnode01 kchuang]# ls -al /etc/docker/certs.d
total 36
drwxr-xr-x. 2 root root 4096 Jan 10 15:02 .
drwx------. 3 root root   53 Mar 10 11:17 ..
-r--------. 1 root root 3326 Jan 10 14:47 ca-key.pem
-r--r--r--. 1 root root 2240 Jan 10 14:49 ca.pem
-rw-r--r--. 1 root root   17 Jan 10 14:59 ca.srl
-r--r--r--. 1 root root 1931 Jan 10 14:59 cert.pem
-rw-r--r--. 1 root root   30 Jan 10 14:58 extfile.cnf
-r--------. 1 root root 3247 Jan 10 14:58 key.pem
-r--r--r--. 1 root root 2086 Jan 10 14:57 server-cert.pem
-r--------. 1 root root 3243 Jan 10 14:51 server-key.pem
```

**Comment by konstruktoid** [comment]
Updated. https://github.com/docker/docker-bench-security/pull/234/commits/6105ff664188eb9e0880090bf99b73a24da3e917

**Comment by kinghuang** [comment]
6105ff6 works for my files!

```
[PASS] 3.9  - Verify that TLS CA certificate file ownership is set to root:root
[PASS] 3.10 - Verify that TLS CA certificate file permissions are set to 444 or more restrictive
[PASS] 3.11 - Verify that Docker server certificate file ownership is set to root:root
[PASS] 3.12 - Verify that Docker server certificate file permissions are set to 444 or more restrictive
[PASS] 3.13 - Verify that Docker server key file ownership is set to root:root
[PASS] 3.14 - Verify that Docker server key file permissions are set to 400 or more restrictive
```

**Comment by konstruktoid** [comment]
And since it fixed #232 as well, I guess it's a merge after some additional testing.
